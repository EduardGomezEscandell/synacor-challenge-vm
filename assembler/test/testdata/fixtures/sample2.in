push r0 text
loop:
    read r8 r0
    out r8

    add r0 r0 1

    eq r1 r0 endtext
    jt loop
halt

text:
    "Hello, world!"
endtext: