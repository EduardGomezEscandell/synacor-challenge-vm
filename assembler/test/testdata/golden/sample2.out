<VERB push> <REGISTER 0> <TAG_REF text> <EOL>
<TAG_DECL loop> <EOL>
<VERB rmem> <REGISTER 8> <REGISTER 0> <EOL>
<VERB out> <REGISTER 8> <EOL>
<VERB add> <REGISTER 0> <REGISTER 0> <NUMBER 1> <EOL>
<VERB eq> <REGISTER 1> <REGISTER 0> <TAG_REF endtext> <EOL>
<VERB jt> <TAG_REF loop> <EOL>
<VERB halt> <EOL>
<TAG_DECL text> <EOL>
<STRING Hello, world!> <EOL>
<TAG_DECL endtext> <EOL>
